### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by mayank99** [sudomf/remix-vite#6]
### Description

Running `npx remix-vite` on a starter remix app in Windows leads to this error:

```console
🖲 remix-vite started at http://localhost:3000
Error: failed to load module for ssr: /@fsC:\Code\remix-vite-test\app\root.tsx
    at instantiateModule (file:///C:/Code/remix-vite-test/node_modules/.pnpm/vite@3.2.2/node_modules/vite/dist/node/chunks/dep-c842e491.js:54154:15)
```

I think the paths are messed up (notice the mismatching forward/back slashes as well as a missing slash).

### Library version

0.1.5

### Node version

18.7.0

**Comment by mayank99** [comment]
I was able to make it run by changing the return of this function:

https://github.com/sudomf/remix-vite/blob/728180419ce097a28791e17aa0abd0766ce3bb38/src/utils/general.ts#L46-L48

to:

```js
return `/@fs/${filePath.replace(/\\/g, '/')}`;
```

but this feels kinda hacky and might break macOS/linux.

---

**Update**: it also doesn't fix the issue. vite says "hmr update" in the terminal but the browser doesn't refresh 😔

I see this error in my browser console:

```console
Uncaught Error: @vitejs/plugin-react can't detect preamble. Something is wrong.
See https://github.com/vitejs/vite-plugin-react/pull/11#discussion_r430879201
    at root.tsx:6:11
```

---

**Update 2**: i was able to move forward with the last error by changing `root.tsx` to remove `<LiveReload>` and include vite refresh scripts.

```diff
export default function App() {
  return (
    <html lang='en'>
      <head>
        <Meta />
        <Links />
      </head>
      <body>
        <Outlet />
        <ScrollRestoration />
+        <ViteScripts />
-        <LiveReload />
        <Scripts />
      </body>
    </html>
  );
}

+ const ViteScripts = () => {
+  return (
+    <>
+      <script type='module' src='/@vite/client' />
+      <script
+        type='module'
+        dangerouslySetInnerHTML={{
+          __html: `import RefreshRuntime from '/@react-refresh'
+   RefreshRuntime.injectIntoGlobalHook(window)
+   window.$RefreshReg$ = () => {}
+   window.$RefreshSig$ = () => (type) => type
+   window.__vite_plugin_react_preamble_installed__ = true`,
+         }}
+       />
+     </>
+   );
+ };

```

**Comment by jrestall** [comment]
I also can't get this working on Windows. There seems to be an issue with the plugins calling `routesByFile.get(id)` that never match due to the routesByFile paths and the id having different representations of the same path.

**Comment by sudomf** [comment]
Hey all. Thanks a lot. I'll definitely focus on this issue ASAP.

**Comment by sudomf** [comment]
> I was able to make it run by changing the return of this function:
> 
> https://github.com/sudomf/remix-vite/blob/728180419ce097a28791e17aa0abd0766ce3bb38/src/utils/general.ts#L46-L48
> 
> to:
> 
> ```js
> return `/@fs/${filePath.replace(/\\/g, '/')}`;
> ```
> 
> but this feels kinda hacky and might break macOS/linux.
> 
> **Update**: it also doesn't fix the issue. vite says "hmr update" in the terminal but the browser doesn't refresh 😔
> 
> I see this error in my browser console:
> 
> ```
> Uncaught Error: @vitejs/plugin-react can't detect preamble. Something is wrong.
> See https://github.com/vitejs/vite-plugin-react/pull/11#discussion_r430879201
>     at root.tsx:6:11
> ```
> 
> **Update 2**: i was able to move forward with the last error by changing `root.tsx` to remove `<LiveReload>` and include vite refresh scripts.
> 
> ```diff
> export default function App() {
>   return (
>     <html lang='en'>
>       <head>
>         <Meta />
>         <Links />
>       </head>
>       <body>
>         <Outlet />
>         <ScrollRestoration />
> +        <ViteScripts />
> -        <LiveReload />
>         <Scripts />
>       </body>
>     </html>
>   );
> }
> 
> + const ViteScripts = () => {
> +  return (
> +    <>
> +      <script type='module' src='/@vite/client' />
> +      <script
> +        type='module'
> +        dangerouslySetInnerHTML={{
> +          __html: `import RefreshRuntime from '/@react-refresh'
> +   RefreshRuntime.injectIntoGlobalHook(window)
> +   window.$RefreshReg$ = () => {}
> +   window.$RefreshSig$ = () => (type) => type
> +   window.__vite_plugin_react_preamble_installed__ = true`,
> +         }}
> +       />
> +     </>
> +   );
> + };
> ```

Can you provide a reproducible example? Or just the `root.tsx` at least. Cause the runtime scripts related to HMR should be auto-injected.

**Comment by jrestall** [comment]
For me I found the plugins aren't working due to the id and routesByFile not matching since they use different slashes in the paths. Looks to be working now that I'm calling path.normalize before passing the id to routesByFile.get. This is in addition to the `/@fs/${filePath}` fix.

e.g. original values that don't match
`id: 'C:/dev/my-app/app/root.tsx'
routesByFile[0]: 'C:\dev\my-app\app\root.tsx'`

working code
`const route = routesByFile.get(path.normalize(id)); // 'C:\dev\my-app\app\root.tsx' 👍 `

**Comment by mayank99** [comment]
> Can you provide a reproducible example? Or just the `root.tsx` at least.

I just used the basic starter from `npm create remix`. https://github.com/remix-run/examples/tree/main/_template

You can find a full repo of my workarounds here: https://github.com/mayank99/remix-vite-test

> Cause the runtime scripts related to HMR should be auto-injected.

They were not, which is why i had to add them manually.

**Comment by sudomf** [comment]
Hey guys. Just to say I'm working on it. Hope to get all these problems solved ASAP.

**Comment by sudomf** [comment]
Hey @mayank99 and @jrestall. Just released 0.2.0 fixing Windows issues. Now we support Windows 🎉 https://github.com/sudomf/remix-vite/releases/tag/0.2.0
